src/new.spec.js

```diff
@@ -111,5 +111,55 @@ describe("withStore", () => {
 
       expect(getState().baz).toBe("baz");
     });
+
+    it("are not called when its state/props dependencies do not change", async () => {
+      const baz = jest.fn(({ qux }, { bar }) => bar * qux);
+      const props = { bar: 2, thud: 9 };
+      const { effects, getState, setParentProps } = makeTestInstance(
+        {
+          initialState: () => ({ qux: 1, corge: 4 }),
+          effects: {
+            changeState() {
+              this.state.corge = 8;
+            },
+          },
+          computed: {
+            baz,
+          },
+        },
+        props
+      );
+
+      expect(getState().baz).toBe(2);
+      setParentProps({ thud: 8 });
+      await effects.changeState();
+      expect(getState().baz).toBe(2);
+      expect(baz.mock.calls.length).toBe(1);
+    });
+
+    it("is called when its state/props dependencies change", async () => {
+      const baz = jest.fn(({ bar }, { qux }) => bar + qux + 2);
+      const props = { qux: 2 };
+      const { effects, getState, setParentProps } = makeTestInstance(
+        {
+          initialState: () => ({ bar: 3 }),
+          effects: {
+            changeState() {
+              this.state.bar = 5;
+            },
+          },
+          computed: {
+            baz,
+          },
+        },
+        props
+      );
+
+      setParentProps({ qux: 4 });
+      expect(getState().baz).toBe(9);
+      await effects.changeState();
+      expect(getState().baz).toBe(11);
+      expect(baz.mock.calls.length).toBe(2);
+    });
```

Comment:
This test should be almost exactly the same than the previous one except that we are changing:
1. its state dependency
2. its props dependency